sdcm/sla/libs/sla_utils.py

```diff
@@ -0,0 +1,285 @@
+    MIXED_LOAD = 'mixed'
+    MIN_CPU_UTILIZATION = 97
+    VALID_DEVIATION_PRC = 10
+
+    @staticmethod
+    # pylint: disable=too-many-arguments, too-many-locals
+    def define_read_cassandra_stress_command(role: Role,
+                                             load_type: str,
+                                             c_s_workload_type: str,
+                                             threads: int,
+                                             stress_duration_min: int,
+                                             max_rows_for_read: int = None,
+                                             stress_command: str = STRESS_READ_CMD,
+                                             throttle: int = 20000,
+                                             num_of_partitions=50000000,
+                                             **kwargs):
+        """
+        :param role: Role object
+        :param load_type: cache_only/disk_only/mixed
+        :param c_s_workload_type: latency: with ops restriction - using throttle
+                                or
+                              throughput: no restriction
+        """
+
+        def latency():
+            return '%d throttle=%d/s' % (threads, throttle)
+
+        def throughput():  # pylint: disable=unused-variable
+            return threads
+
+        def cache_only(max_rows_for_read):  # pylint: disable=unused-variable
+            if not max_rows_for_read:
+                max_rows_for_read = int(num_of_partitions * 0.3)
+            return 'seq=1..%d' % max_rows_for_read
+
+        # Read from cache and disk
+        def mixed(max_rows_for_read):  # pylint: disable=unused-variable
+            if not max_rows_for_read:
+                max_rows_for_read = num_of_partitions
+            return "'dist=gauss(1..%d, %d, %d)'" % (max_rows_for_read,
+                                                    int(max_rows_for_read / 2),
+                                                    int(max_rows_for_read * 0.05))
+
+        def disk_only(max_rows_for_read):  # pylint: disable=unused-variable
+            if not max_rows_for_read:
+                max_rows_for_read = int(num_of_partitions * 0.3)
+            return 'seq=%d..%d' % (max_rows_for_read, max_rows_for_read + int(num_of_partitions * 0.25))
+
+        rate = locals()[c_s_workload_type]()  # define -rate for c-s command depend on workload type
+        pop = locals()[load_type](max_rows_for_read)  # define -pop for c-s command depend on load type
+
+        params = {'n': num_of_partitions, 'user': role.name, 'password': role.password, 'pop': pop,
+                  'duration': '%dm' % stress_duration_min, 'threads': rate}
+        if kwargs:
+            params.update(kwargs['kwargs'])
+        c_s_cmd = stress_command.format(**params)
+        LOGGER.info("Created cassandra-stress command: %s", c_s_cmd)
+
+        return c_s_cmd
+
+    @staticmethod
+    def one_run_c_s_stats(read_run, user_name, statistic_name, tester):
```

Comment:
The method name isn't articulated well in my opinion.

Reply:
And I would add a comment to why it's needed. and what are the stats.

Reply:
renamed and added a comment